src/builder/QuarkBuilder.sol

```diff
@@ -242,7 +237,13 @@ contract QuarkBuilder {
                             chainAccountsList[i].chainId, transferIntent.chainId, transferIntent.assetSymbol
                         )
                 ) {
-                    aggregateTransferAssetAvailableBalance += Accounts.sumBalances(positions);
+                    // If the payment token is the transfer token and user opt for paying with the payment token, reduce the available balance by the maxCost
+                    if (payment.isToken && Strings.stringEqIgnoreCase(payment.currency, transferIntent.assetSymbol)) {
+                        uint256 maxCost = PaymentInfo.findMaxCost(payment, chainAccountsList[i].chainId);
+                        aggregateTransferAssetAvailableBalance += Accounts.sumBalances(positions) - maxCost;
+                    } else {
+                        aggregateTransferAssetAvailableBalance += Accounts.sumBalances(positions);
+                    }
```

Comment:
For available fund when any actions (all actions will cost gas) to be applied onto it when token is payment token and user opt to pay with token.. Will always be balance - maxCost.  
So if we know user has 2USDC mainner (0.1 maxCost), and 2USDC on Base (0.01maxCost) and user request to transfer 3.99 USDC.  We can simply check (2 + 2 - 0.1 - 0.01 = 3.89) and failed user request, as we are sure with actions apply on to the quarkbuilder even with bridging,  it won't be enough.